frontend/src/v5/ui/routes/dashboard/projects/federations/editFederationModal/editFederation/editFederationContainersList/editFederationContainersListItem/editFederationContainersListItemLoading.component.tsx

```diff
@@ -0,0 +1,33 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ContainersActionsDispatchers } from '@/v5/services/actionsDispatchers';
+import { TeamspacesHooksSelectors, ProjectsHooksSelectors } from '@/v5/services/selectorsHooks';
+import { SkeletonListItem } from '@components/shared/revisionDetails/components/skeletonListItem/skeletonListItem.component';
+import { useEffect } from 'react';
+
+export const EditFederationContainersListItemLoading = ({ index, container }) => {
+	const teamspace = TeamspacesHooksSelectors.selectCurrentTeamspace();
+	const project = ProjectsHooksSelectors.selectCurrentProject();
+	
+	useEffect( () => {
+		ContainersActionsDispatchers.fetchContainerStats(teamspace, project, container._id);
```

Comment:
I think this might result in some unnecessary fetchContainerStats calls, but I don't understand why...
Removing this causes the items to be stuck in the loading stage despite the fact the stats are fetched in fetchContainers beforehand.
Also if a container is favourited then this will get called twice. Once for the top list, and once for the bottom list

Reply:
It actually seems to get called twice per item in the top list. And then once again per item in bottom list. I think this might be why it's quite slow when opening the edit model for a federations with 1000 containers